src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,205 @@
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.model.request.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfigItem;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.CallbackException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.payments.PaymentsPublishingException;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import java.util.Map;
+
+import static java.lang.String.format;
+
+@Service
+public class CcdCaseUpdater {
+    private static final Logger log = LoggerFactory.getLogger(CcdCaseUpdater.class);
+
+    private static final String EXCEPTION_RECORD_REFERENCE = "bulkScanCaseReference";
+
+    private final CaseUpdateClient caseUpdateClient;
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdCaseUpdater(
+        CaseUpdateClient caseUpdateClient,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.caseUpdateClient = caseUpdateClient;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
+    public void updateCase(
+        ExceptionRecord exceptionRecord,
+        ServiceConfigItem configItem,
+        String idamToken,
+        String userId,
+        CaseDetails existingCase
+    ) {
+        log.info(
+            "Start updating case for {} with case ID {} from exception record {}",
+            configItem.getService(),
+            existingCase.getId(),
+            exceptionRecord.id
+        );
+
+        try {
+            String s2sToken = s2sTokenGenerator.generate();
+
+            SuccessfulUpdateResponse updateResponse = caseUpdateClient.updateCase(
+                configItem.getUpdateUrl(),
+                existingCase,
+                exceptionRecord,
+                s2sToken
+            );
+
+            log.info(
+                "Successfully updated case for {} with case ID {} from exception record {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            checkBulkScanReferenceIsSet(
+                (Map<String, ?>) updateResponse.caseDetails.caseData,
+                exceptionRecord.id
+            );
```

Comment:
There's no requirement for this field to be set in the update endpoint. I don't think this should happen at all. `bulkScanCaseReference` is used for preventing the creation of multiple cases from a single exception record. Update should be idempotent - doesn't matter how many times we update the case with the same exception record - the resulting case data should be the same.

Reply:
`bulkScanCaseReference` just should be set to the right value, I think we should keep this check.

Reply:
But it's a different endpoint that sets it - this reference means `the ID of the exception record from which the case was created`. The name of this field isn't great, but that's what it is. It shouldn't be changed to the ID of the exception record that updated the case.